src/transformers/modeling_dpr.py

```diff
@@ -0,0 +1,551 @@
+            They correspond to the combined `input_ids` from `(question + context title + context content`).
+        **start_logits**: ``torch.FloatTensor`` of shape ``(n_passages, sequence_length)``
+            Logits of the start index of the span for each passage.
+        **end_logits**: ``torch.FloatTensor`` of shape ``(n_passages, sequence_length)``
+            Logits of the end index of the span for each passage.
+        **relevance_logits**: `torch.FloatTensor`` of shape ``(n_passages, )``
+            Outputs of the QA classifier of the DprReader that corresponds to the scores of each passage
+            to answer the question, compared to all the other passages.
+
+
+    Examples::
+
+        tokenizer = DprReader.from_pretrained('dpr-base-uncased')
+        model = DprModel.from_pretrained('dpr-reader-base')
+        question_and_titles_ids = [
+            torch.tensor(tokenizer.encode("Hello, is my dog cute ?", "Dog cuteness"))
+            ]  # One tensor per passage. It corresponds to the concatenation of the question and the context title.
+        texts_ids = [
+            torch.tensor(tokenizer.encode("Hello, my dog is definitely cute", add_special_tokens=False))
+            ]  # One tensor per passage. It corresponds to the context text in which we're looking for the answer.
+        outputs = model(question_and_titles_ids, texts_ids)
+        start_logits = outputs[1]  # The logits of the start of the span
+        end_logits = outputs[2]  # The logits of the end of the span
+
+    """
+
+    def __init__(self, config: DprConfig):
+        super().__init__(config)
+        self.config = config
+        self.reader = get_bert_reader_component(config)
+        self.init_weights()
+
+    def forward(self, question_and_titles_ids: List[T], texts_ids: List[T],) -> Tuple[T, ...]:
+        assert len(question_and_titles_ids) == len(texts_ids)
+        device = question_and_titles_ids[0].device
+        n_contexts = len(question_and_titles_ids)
+        input_ids = torch.ones((n_contexts, self.config.sequence_length), dtype=torch.int64) * int(self.config.pad_id)
+        input_ids = input_ids.to(device=device)
+        for i in range(n_contexts):
+            question_and_title_ids = question_and_titles_ids[i]
+            text_ids = texts_ids[i]
+            _, len_qt = question_and_title_ids.size()
+            _, len_txt = text_ids.size()
+            input_ids[i, 0:len_qt] = question_and_title_ids
+            input_ids[i, len_qt : len_qt + len_txt] = text_ids
+        input_ids = input_ids.unsqueeze(0)
+        attention_mask = input_ids != self.config.pad_id
+        attention_mask = attention_mask.to(device=device)
+        start_logits, end_logits, relevance_logits = self.reader(input_ids, attention_mask)
+        return input_ids, start_logits, end_logits, relevance_logits
+
+    def generate(
+        self,
+        question_and_titles_ids: List[T],
+        texts_ids: List[T],
+        k: int = 16,
```

Comment:
not a fan of single letters `k` => very hard to understand what it is for people

Reply:
this one I agree

Reply:
What about `num_spans` ?